### DISCUSSION TITLE
How to provide a custom string ID for class registration rather than int ID?

### DISCUSSION THREAD

**Discussion by Rubydesic** [apache/fory#1969] [Q&A] [answered]
Clearly, Fury supports using String IDs for classes, since it serializes the fully qualified name of unregistered classes when registration is not required. However, I can't find any way to manually set the key/ID as a string. 

This is useful because I want to get the security benefit of requiring classes to be registered while having the flexibility to use any string as the ID for a class.

**Comment by chaokunyang** [comment]
Fury doesn't support to register a string name for class. We do have plans for this, But not this month. Currently we are working on xlang serialization. This feature may be implemented in next two months

**Reply by Rubydesic** [reply]
thank you for your answer

**Comment by chaokunyang** [comment, marked as answer]
Hi @Rubydesic , this feature is implemented in https://github.com/apache/fury/pull/2053, you can try it out using the snapshot version of fury

**Reply by Rubydesic** [reply]
Awesome, thank you.
